<launch>
	<!-- Set the arguments -->
	<arg name="num" default="12"/>  <!-- Number of robots -->
	<arg name="map_name" default="empty_10x10"/>  <!-- Name of the simulation map -->
	
	<arg name="map_yaml" default="$(find sphero_formation)/sim/$(arg map_name).yaml"/>

	<!-- Include launch file with Reynolds controllers -->
	<include file="$(find sphero_formation)/launch/real/$(arg num)_robots.launch"/>

	<!-- Start nearest_search node: search for other Spheros in range-->
	<node pkg="sphero_formation" type="nearest_search.py" name="search" output="screen">
		<param name="num_of_robots" type="int" value="$(arg num)"/>
	</node>

	<!-- Start simulation_tf node: provide tf transforms for simulation-->
<!-- 	<node pkg="sphero_formation" type="simulation_tf.py" name="tf_server"/>
 -->

	<!-- Start map serever -->
	<node pkg="map_server" type="map_server" args="$(arg map_yaml)" name="map_server"/>
</launch>